test/utils/browser.js

```diff
@@ -30,32 +52,83 @@ exports.openMusic = function (options) {
   // Execute many async steps
   before(function startBrowser () {
     this.browser = wd.remote();
+    // Uniquely identify a browserstack build based on environment variable
+    if (process.env.USE_BROWSERSTACK) {
+      this.browser = wd.remote('hub.browserstack.com', 80);
+    }
     global.browser = this.browser;
   });
   before(function openBrowser (done) {
-    this.browser.init({browserName: 'chrome'}, done);
+    var that = this;
+    // DEV: Force the Windows 8 platform as it appears to run our test more
+    //      stable than other platforms
+    this.browser.init({
+      browserName: 'chrome',
+      name: 'google-music.js - ' + this.test.parent.title,
```

Comment:
Isn't it physically impossible to run a before outside of a containing describe?  It could be a parameter but then all the containing describes would have to change which would look even more magical?